### PR TITLE
53 set authentication in database

### PR DESCRIPTION
**Pull request by AlbertoPaulPopescu**
Closes https://github.com/UserOfficeProject/stfc-user-office-project/issues/53?fbclid=IwAR0xrSDaa7DZGVzB0XktksawLKN6SzI-tiUpvioRK9Ls3c9rdihAxfyAcik

The authentication link and status is set in the DB and called at runtime now. The 'Features' table contains the EXTERNAL_AUTH state. The 'Settings' table contains the EXTERNAL_AUTH_LOGIN_URL. The 'process.env.REACT_APP_AUTH_TYPE' and 'process.env.REACT_APP_EXTERNAL_AUTH_LOGIN_URL' was replaced with 'EXTERNAL_AUTH' and 'EXTERNAL_AUTH_LOGIN_URL', respectively, with one exception (App.tsx line 121 - missed).

I had to create a different new pull request for this issue as there were many changes that happened since the first pull request was created and I was receiving multiple errors when working on the old branch: https://github.com/UserOfficeProject/user-office-frontend/compare/53-move-auth-variable

1: The settings table was created in the database.
2: 'externalAuth' feature was inserted in the Features table.
3: The code was written following the 'Features' example.
4: Authenticating is fully working 

Known issues: 
1: Had an issue with the 'useDataApi.ts' file, line 59 where I replaced: 'error.response.errors[0].message' with 'error.response.error' to bypass the error and test my changes. However, my fix was reverted back to 'error.response.errors[0].message' before creating the PR.
2: There are still linter errors in the backend but I just followed the already existent convention in the code.
3: In App.tsx I need to replace in line 121 with EXTERNAL_AUTH === true which I missed.

Would like to know:
1: Is the naming convention used fine, any changes/suggestions?
2: Is there any part of the code that could be improved?
3: Is there any old code that needs to be removed as part of this issue?

How Has This Been Tested

1: The new ESS app authentication works as expected when logged in or not logged in into the old P&O system.
2: Tested on Google Chrome.

## Tests included/Docs Updated?

1: An unit test was added to certify data is retrieved from the database.

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

**Review by jekabs-karklins — APPROVED** [review]
Need to fix the build errors, but otherwise looks good

**Comment by AlbertoPaulPopescu** [comment]
> Need to fix the build errors, but otherwise looks good

@jekabs-karklins  Thank you for reviewing, I just fixed (almost) my docker and I'll hopefully fix it today.

**Comment by AlbertoPaulPopescu** [comment]
How could I use 'useContext' in 'render()' within the 'class App' (line: 83)? Thanks!
image

**Comment by jekabs-karklins** [comment]
@AlbertoPaulPopescu don't forget to merge it

File: src/components/App.tsx [hunk 1]
```diff
@@ -44,11 +57,10 @@ const PrivateRoute: React.FC<RouteProps> = ({ component, ...rest }) => {
           render={(props): JSX.Element => {
             if (!token) {
               if (
-                process.env.REACT_APP_AUTH_TYPE === 'external' &&
-                process.env.REACT_APP_EXTERNAL_AUTH_LOGIN_URL
+                EXTERNAL_AUTH === true &&
```
Comment (JEThompson97) [inline src/components/App.tsx:60]: Could these be changed to just
```suggestion
                EXTERNAL_AUTH &&
```

Comment (AlbertoPaulPopescu) [reply src/components/App.tsx]: I changed this as per your suggestion in the commit below.

**Review by JEThompson97 — APPROVED** [review]
Nice work getting this one sorted!

**Pull request merged**